project added

Viktoria8888 opened the pull request.

Viktoria8888 committed:
project added

porfinogeneta commented on src/ZosiaSection.tsx:
funkcyjne komponenty React'owe robimy bez tego React.FC, jako zwykłą funkcję

porfinogeneta commented on src/ZosiaSection.tsx:
```diff
@@ -0,0 +1,28 @@
+                    backgroundSize: "contain",
+                    backgroundPosition: "center top",
+                    backgroundRepeat: "no-repeat",
+                    height: "25vh",
```
raczej podawać wielkości w tailwindzie, a nie tak, bo wtedy prościej uzyskać responsywaność, zatem stylowanie przerobić na tailwind

porfinogeneta commented on src/ZosiaSection.tsx:
```diff
@@ -0,0 +1,28 @@
+
+
+            <div
+                className="absolute top-10 left-0 right-0 bg-no-repeat bg-contain bg-center opacity-20"
```
zrobić flexboxa

porfinogeneta commented on src/ZosiaSection.tsx:
```diff
@@ -0,0 +1,28 @@
+
+            <div className="relative z-10 pt-[32vh] p-10 text-center">
+                <h2 className="text-dark-text-primary text-4xl ">Nasze Projekty</h2>
+                <div className="w-[811px] h-1 bg-action-blue mx-auto mt-2"></div>
```
arbitralne wartości lepiej zastąpić czymś co będzie się skalowało, jak już muszą być konkretne liczby, to lepiej skorzystać z liczb wbudowanych w tailwind

porfinogeneta commented on src/Projects.tsx:
```diff
@@ -0,0 +1,38 @@
+];
+
+
+const Projects: React.FC = () => {
```
inaczej robimy komponenty reactowe

porfinogeneta commented on src/ProjectCard.tsx:
```diff
@@ -0,0 +1,28 @@
+const ProjectCard: React.FC<ProjectProps> = ({ title, description, image, color, skewAmount = 0 }) => {
+  return (
+    <section className={`${color} z-10 relative pt-20 pb-20 flex justify-center items-center`}
+      style={{ clipPath: `polygon(0 ${10 - skewAmount}%, 100% 0, 100% ${90 + skewAmount}%, 0 100%)` }}>
```
stylowanie zamienić na tailwind

porfinogeneta commented on src/ProjectCard.tsx:
```diff
@@ -0,0 +1,28 @@
+          <p className="text-dark-text-secondary">{description}</p>
+        </div>
+        <div>
+          <img src={image} alt="zosia" className="rounded-lg min-w-[210px] aspect-square object-cover" />
```
pozbyć się losowych liczb

porfinogeneta commented on src/Projects.tsx:
wpasować się w podział projektu, widoczny w branchu feature-main-page

porfinogeneta commented on src/ZosiaSection.tsx:
```diff
@@ -0,0 +1,28 @@
+const ZosiaSection: React.FC = () => {
+    return (
+        <div className="relative bg-dark-background-primary h-[60vh]  text-white flex flex-col items-center">
```
skorzystać z tailwindowych wartości, zrezygnować z absolute

porfinogeneta commented on package.json:
```diff
@@ -5,28 +5,29 @@
-    "build": "tsc -b && vite build",
-    "lint": "eslint .",
+    "build": "tsc && vite build",
+    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
```
raczej nie chcemy configa zmieniać

porfinogeneta commented on package.json:
```diff
@@ -5,28 +5,29 @@
-    "typescript": "~5.6.2",
-    "typescript-eslint": "^8.18.2",
-    "vite": "^6.0.5"
+    "@types/react": "^18.2.48",
```
wziąć package.json z master albo z feature-main-page

porfinogeneta reviewed (commented):
generalnie ok, tylko flexbox gdzie się tylko da, wywalić absoluty, korzystać ze stylowania w tailwind i korzystać z tailwindowych stałych + dodać responsywność

czarekmilek committed:
Merge branch 'master' into projects

czarekmilek committed:
feat (frontend): Added `Projects` page routing

czarekmilek committed:
refactor (frontend): Moved `Projects` related components into proper folder

czarekmilek committed:
fix (frontend): Fixed typo in "HeroSection" component

czarekmilek committed:
feat (frontend): Improved styling

- Added Zosia logo background as an adjacent component
- Adjusted styling of all `Project` related components

czarekmilek committed:
feat (frontend): Extracted projects data into adjacent file

czarekmilek committed:
refactor (frontend): Used grid instead of flex for project card

czarekmilek committed:
feat (frontend): Added polygons based project card design

czarekmilek committed:
feat (frontend): Rearranged project card scheme

- Odd cards have text content on the left and image on the right
- Even cards have image on the left and text on the right

czarekmilek committed:
feat (frontend): Made design more mobile friendly

czarekmilek committed:
feat (frontend): Added read more button for bigger projects

czarekmilek commented:
Poprawki wykonane. Poza tym, na co zwróciłeś uwagę, dorobiłem kilka dodatkowych rzeczy.
